snapshot-labs/sx-monorepo
# feat: user profile page

### Summary



Closes: #387

> This PR is bigger than previous ones, due to including the whole feature in a single PR, instead of breaking them down in multiple PR like previously.

This feature introduce the "User" concept for real, with an usable User profile page, as well as editable profile details.

### Changes

- The profile link in the modal account is now redirecting to the user profile page, instead of etherscan

![Screenshot 2024-06-11 at 01 09 58](https://github.com/snapshot-labs/sx-monorepo/assets/495709/26626107-88ea-45bb-93ce-aaa6a2b3080b)

- User profile page now has content. As of now, it only show activities from onchains spaces, as offchain leaderboard is not ready yet

![Screenshot 2024-06-11 at 01 10 43](https://github.com/snapshot-labs/sx-monorepo/assets/495709/aeebb08e-b4a0-4e46-b8ec-8f35c1788892)

- User profile data are now editable (saved to offchain network), via the edit profile modal

![Screenshot 2024-06-11 at 01 11 36](https://github.com/snapshot-labs/sx-monorepo/assets/495709/e1237492-7629-4327-9950-daefa6040d84)

- The user menu in the page top right is now showing the user's name (as set in the profile) when available
- Offchain network is now the only source of truth for user metadata (name, about, avatar, etc ...)
- UI wise, it follows the existing layout from Space overview for the header, and Leaderboard for the activity table, instead of following the figma, which is using different font/line weight and spacing, text alignment, etc ... for more consistency
- Updating the user profile is done via alias, to avoid signing a transaction each time, similar to follow/unfollow space
- Checking for profile will not fail if the user does not exist in the database, as long as the address is valid (since onchain users do not exist in the offchain database)
- A new "Copy link" option has been added to the share dropdown, and has been extracted to its own component for reusability

![Screenshot 2024-06-11 at 01 13 11](https://github.com/snapshot-labs/sx-monorepo/assets/495709/2122df65-2cae-4439-9742-8f9b4b73043f)

- User profile page now have the sidebar from the "My" section (will have the "active" class only if it's your own profile)
- "My" section sidebar does not require to be logged in anymore, and will display different content depending on connection status: only show "Explore" for guest user
- `UiInputStampCover` component has been refactored to also support user cover
- User cover will default to use the blur avatar when not set, similar to space

### How to test

1. Access your profile via the account modal
2. It should show your profile
3. You can edit your profile via the Edit profile modal
4. Saving should ask for signing an alias if not already done, and just save, and does not ask for a signature.
5. Your profile should be updated with the newly edited data

### TODO 

- [x] Fetch user cover from stamp (waiting for https://github.com/snapshot-labs/stamp/pull/240) - wrong cover image may be shown until the pr is merged
- [x] Show user's username in topnav account menu
- [ ] ~~Support starknet user on offchain~~ (will be added in another PR)
- [x] Get a better message for the social share
